docs/reference/nanodet.md

```diff
@@ -0,0 +1,289 @@
+## nanodet module
+
+The *nanodet* module contains the *NanodetLearner* class, which inherits from the abstract class *Learner*.
+
+### Class NanodetLearner
+Bases: `engine.learners.Learner`
+
+The *NanodetLearner* class is a wrapper of the Nanodet object detection algorithms based on the original
+[Nanodet implementation](https://github.com/RangiLyu/nanodet).
+It can be used to perform object detection on images (inference) and train All predefined Nanodet object detection models and new modular models from the user.
+
+The [NanodetLearner](../../src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py) class has the
+following public methods:
+
+#### `NanodetLearner` constructor
+```python
+NanodetLearner(self, model_to_use, iters, lr, batch_size, checkpoint_after_iter, checkpoint_load_iter, temp_path, device,
+               weight_decay, warmup_steps, warmup_ratio, lr_schedule_T_max, lr_schedule_eta_min, grad_clip)
+```
+
+Constructor parameters:
+
+- **model_to_use**: *{"EfficientNet-Lite0_320", "EfficientNet-Lite1_416", "EfficientNet-Lite2_512", "RepVGG-A0_416",
+  "t", "g", "m", "m-416", "m-0.5x", "m-1.5x", "m-1.5x-416", "plus-m_320", "plus-m-1.5x_320", "plus-m_416",
+  "plus-m-1.5x_416", "custom"}*\
+  Specifies the model to use and the config file that contains all hyperparameters for training, evaluation and inference as the original
+  [Nanodet implementation](https://github.com/RangiLyu/nanodet). If you want to overwrite some of the parameters you can
+  put them as parameters in the learner.
+- **iters**: *int, default=None*\
+  Specifies the number of epochs the training should run for.
+- **lr**: *float, default=None*\
+  Specifies the initial learning rate to be used during training.
+- **batch_size**: *int, default=None*\
+  Specifies number of images to be bundled up in a batch during training.
+  This heavily affects memory usage, adjust according to your system.
+- **checkpoint_after_iter**: *int, default=None*\
+  Specifies per how many training iterations a checkpoint should be saved.
+  If it is set to 0 no checkpoints will be saved.
+- **checkpoint_load_iter**: *int, default=None*\
+  Specifies which checkpoint should be loaded.
+  If it is set to 0, no checkpoints will be loaded.
+- **temp_path**: *str, default='temp'*\
```

Comment:
In the code the default value is
```suggestion
- **temp_path**: *str, default=''*\
```

Reply:
Thank you for your suggestion. Also add comment on what will hapend if the user use the default value.